WordPress/Classes/Services/MediaService.m

```diff
@@ -30,71 +60,65 @@ - (void)createMediaWithPHAsset:(PHAsset *)asset
         }
         return;
     }
-    MediaType mediaType = MediaTypeDocument;
-    NSSet *allowedFileTypes = nil;
+    
+    MediaType mediaType = [asset assetMediaType];
     NSString *assetUTI = [asset originalUTI];
     NSString *extension = [self extensionForUTI:assetUTI];
-    if (asset.mediaType == PHAssetMediaTypeImage) {
-        mediaType = MediaTypeImage;
-        allowedFileTypes = post.blog.allowedFileTypes;
+    if (mediaType == MediaTypeImage) {
+        NSSet *allowedFileTypes = post.blog.allowedFileTypes;
         if (![allowedFileTypes containsObject:extension]) {
             assetUTI = (__bridge NSString *)kUTTypeJPEG;
             extension = [self extensionForUTI:assetUTI];
         }
-    } else if (asset.mediaType == PHAssetMediaTypeVideo) {
+    } else if (mediaType == MediaTypeVideo) {
         /** HACK: Sergio Estevao (2015-11-09): We ignore allowsFileTypes for videos in WP.com
-         because we have an exception on the server for mobile that allows video uploads event 
+         because we have an exception on the server for mobile that allows video uploads event
          if videopress is not enabled.
-        */
-        if (![post.blog isHostedAtWPcom] && ![allowedFileTypes containsObject:extension]) {
```

Comment:
Good catch, we actually have a little glitch on this code, in the future we may need to check for self-hosted sites what video formats they support, but for now this is ok.